### PR TITLE
Make some non-unique ids unique for input fields. For fields whose id…

### PR DESCRIPTION
**Pull request by engineSound**
…s are hard to remove, just delete the ids.

### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [x] Automated Testing

- [x] Sufficient User and Developer Documentation

- [x] Passing Jenkins Build

- [x] Peer Code review and approval

### Positive Test Case
Open chrome browser developer mode, navigate to console. Watch errors panel. Make sure there is no error reported in the following steps.
1. Have a resource that have comments.
2. Visit the resource landing page.
3. Edit the resource landing page.

### CODE REVIEW

File: hs_core/views/__init__.py [hunk 1]
```diff
@@ -1091,14 +1091,27 @@ class AddUserForm(forms.Form):
...
                                        widget=autocomplete_light.ChoiceWidget("GroupAutocomplete"))
 
     return {
-        'add_owner_user_form': AddUserForm(),
+        'add_view_contri_user_form': AddUserContriForm(),
+        'add_view_invite_user_form': AddUserInviteForm(),
+        'add_view_hs_user_form': AddUserHSForm(),
         'add_view_user_form': AddUserForm(),
-        'add_edit_user_form': AddUserForm(),
         'add_view_group_form': AddGroupForm(),
```
Comment (aphelionz) [inline hs_core/views/__init__.py]: Do we need to do he same for AddGroupForm? or is that ok?

Comment (engineSound) [reply hs_core/views/__init__.py]: good catch. Currently there is only one active use of add_view_group_form. So we don't have dup id coming out yet. But if someone reuse the same template, there will be duplicate ids.

Comment (aphelionz) [reply hs_core/views/__init__.py]: Leave a comment there and let people know of the risk?

Comment (aphelionz) [reply hs_core/views/__init__.py]: Please add a comment at this point in the code so future devs know of the danger

**Comment by Maurier** [comment]
Looks good. Not getting any of the id errors we got before.
+1

**Comment by engineSound** [comment]
@Maurier Can you take a look one more time? I think there any conflict with your resource landing page redesign theme/templates/resource-landing-page/modals.html? If so, I would hold off the part that overlaps with yours.

**Comment by Maurier** [comment]
@engineSound I will come back to this once the Resource landing page PR is merged and solve the merge conflicts as I am more familiar with that work.

**Comment by hydrobuild** [comment]
Can one of the admins verify this patch?

**Comment by Maurier** [comment]
@engineSound I have updated this branch with the latest work in the develop branch. I had to update some of the HTML and scripts to do so. I tested and made sure the functionality works. Please give it a test yourself and confirm.

**Comment by engineSound** [comment]
@Maurier to clarify, you have updated the 2709_nonUniq_id branch?

**Comment by engineSound** [comment]
test this please

File: theme/templates/resource-landing-page/subject.html [hunk 2]
```diff
@@ -40,8 +40,8 @@ <h3>Subject Keywords</h3>
                     <fieldset>
                         <div id="div_id_value" class="control-group">
                             <div class="controls">
-                                <input class="form-control input-sm textInput" id="id_value" maxlength="500"
-                                       name="value"
+                                <input class="form-control input-sm textInput" id="id_subject_keyword_control_input"
```
Comment (Maurier) [inline theme/templates/resource-landing-page/subject.html]: @sblack-usu This is the conflicting part with our scripts.

**Comment by engineSound** [comment]
@Maurier Which scripts the id has conflict with?

**Comment by engineSound** [comment]
@Maurier @sblack-usu That the keywords is not saved, is caused by the id-mismatch between subject.html and resourceLandingAjax.js and generic-resources.js. The id in the html has changed from "id_value" to "id_subject_keyword_control_input", the script still refers "id_value". I am going to create a github issue and submit a PR today.

**Comment by Maurier** [comment]
@engineSound, This is already taken care of. See https://github.com/hydroshare/hydroshare/pull/3210

**Pull request merged**